src/editor/editorui.js

```diff
@@ -60,10 +54,93 @@ export default class EditorUI {
+			 * {@link module:core/editor/editorui~EditorUI#constructor EditorUI constructor}. Only subclass (for example
+			 * {@link module:editor-classic/classiceditorui~ClassicEditorUI}) should use it without passing it further.
+			 *
+			 * @error deprecated-editorui-view-param-in-constructor
+			 */
+			log.warn( 'deprecated-editorui-view-param-in-constructor: The EditorUI#constructor `view` parameter is deprecated.' );
+		}
+
 		// Informs UI components that should be refreshed after layout change.
 		this.listenTo( editor.editing.view.document, 'layoutChanged', () => this.update() );
 	}
 
+	/**
+	 * **Deprecated** since `v12.0.0`. This property is deprecated and should not be used. Use the property
+	 * from the subclass directly instead, for example
+	 * {@link module:editor-classic/classiceditorui~ClassicEditorUI#view ClassicEditorUI#view}.
+	 *
+	 * The main (top–most) view of the editor UI.
+	 *
+	 * @deprecated v12.0.0 This property is deprecated and should not be used. Use the property
+	 * from the subclass directly instead, for example
+	 * {@link module:editor-classic/classiceditorui~ClassicEditorUI#view ClassicEditorUI#view}.
+	 * @readonly
+	 * @member {module:ui/editorui/editoruiview~EditorUIView} #view
+	 */
+	get view() {
+		/**
+		 * This error is thrown when a component tries to access deprecated
+		 * {@link module:core/editor/editorui~EditorUI#element `EditorUI view`} property. Instead the `view` property
+		 * from the subclass (for example {@link module:editor-classic/classiceditorui~ClassicEditorUI#view ClassicEditorUI#view})
+		 * should be accessed directly.
+		 *
+		 * @error deprecated-editorui-view
+		 */
+		log.warn( 'deprecated-editorui-view: The EditorUI#view property is deprecated.' );
+
+		return this._view;
+	}
+
+	/**
+	 * The main (outermost) DOM element of the editor UI.
+	 *
+	 * For example, in {@link module:editor-classic/classiceditor~ClassicEditor} it is a `<div>` which
+	 * wraps the editable element and the toolbar. In {@link module:editor-inline/inlineeditor~InlineEditor}
+	 * it is the editable element itself (as there is no other wrapper). However, in
+	 * {@link module:editor-decoupled/decouplededitor~DecoupledEditor} it is set to `null` because this editor does not
+	 * come with a single "main" HTML element (its editable element and toolbar are separate).
+	 *
+	 * This property can be understood as a shorthand for retrieving the element that a specific editor integration
+	 * considers to be its main DOM element.
+	 *
+	 * @readonly
+	 * @member {HTMLElement|null} #element
+	 */
+	get element() {
+		return null;
+	}
+
+	/**
+	 * Fires the {@link module:core/editor/editorui~EditorUI#event:ready `ready`} event.
+	 *
+	 * This method should be called after {@link module:core/editor/editor~Editor#event:pluginsReady} and before
+	 * {@link module:core/editor/editor~Editor#event:dataReady} by the specific editor creator.
+	 */
+	ready() {
+		this.fire( 'ready' );
+	}
```

Comment:
I am not a big fan of this method. I think that specific *editorui` classes should fire `ready` when... they are ready. Most probably at the end of the `init` method. However, I do not like the fact that this event is triggered be an external class.

Reply:
This method is a shortcut/proxy for `editor.ui.fire( 'ready' )`. It is not fired anywhere in this class, the [specific creator decides when to fire it](https://github.com/ckeditor/ckeditor5-editor-classic/blob/442b578e2a725a1bba9b5972668a75aaedcf76ea/src/classiceditor.js#L200). So it is the matter of creator executing `editor.ui.fire( 'ready' )` or `editor.ui.ready()` code.

And as you pointed out in https://github.com/ckeditor/ckeditor5-editor-classic/pull/77#discussion_r248342654 we could modified this method to fire both `ready` and `uiReady` which gives more flexibility.

Reply:
As we talked F2F, `ready` event could be fired in the `init` method in each `*EditorUI` class.

Reply:
I think it would be better if the base editor constructor bind `editor.uiReady` event to `editor.ui.ready`  event. Then `*EditorUI` could just `this.fire( 'ready' );` in init, instead of calling this method. I think that there should be no such method in this class API, because it has no usage for other classes.

Reply:
> I think that there should be no such method in this class API, because it has no usage for other classes.

Ok, I got the point, now it is called only by the instances of the class itself and no external classes, so it should be protected/private or removed as you suggested. Before we moved it here it was called in `Editor` which made sense that it was a public method but now it doesn't 👍 

As for binding `uiReady` to `ready` event, I'm ok with that, but not sure how it should be done (technically). Do we have an API for binding one vente to another so when one is triggered the second gets triggered too?

Reply:
Yes. You can use [delegate](https://ckeditor.com/docs/ckeditor5/latest/api/module_utils_emittermixin-Emitter.html#function-delegate).